### ISSUE TITLE
possible issue with _test environment not being rebuilt correctly

### ISSUE THREAD

**Issue by davidslac** [conda/conda-build#1193]
I am having an issue where my _test environment is dirty. I'm going through a development cycle where I try to build my package, debug things by either doing conda-build with the --test option, or manually activating the _test environment, and doing things like install ipython and debug, modify the conda-build.sh that is in place and re-run it. After running my latest conda-build, I can see that the a shared object library file from my package, in the _test/lib directory is out of date, while a file in _test/lib/python2.7/site-packages/ is, I think more current - honestly this may be my problem - it may be that my build step has a problem with installing files - I don't have a small example to share at this point - however, I thought I would raise the issue in case you wanted to look into it.

**Comment by msarahan** [comment]
Thanks for raising the issue.  the --test option is not very well tested right now, and a bug may certainly exist.  Unfortunately, I really need to press on with other things right now, rather than bug-hunt.  If you can isolate it further, please let us know.  It may go away on its own anyway, with refactoring work being done for conda-build 2.0.

**Comment by davidslac** [comment]
No worries, I removed the _test env, and re-did the conda build. One thing with my conda build, I'm not bumping up the build number, I'm recreating the package, I'm expecting the pkg cache file to change when done. Even though I removed _test (I tried to do conda env remove, but ran into this issue https://github.com/conda/conda/issues/3251) after the test section of the meta.yaml fails, and I go into the _test environment, my installed files has the inode for an older version in the .pkgs subdir. It is weird, I'm sure my install is copying in the updated file.

Also, it seems that since I have a test section in my meta.yaml, the _build environment is removed. I think it would be helpful to have the _build environment around.

Another comment about the test section - I think it had some of the environment variables defined that I set in my build.sh - maybe I messed something up but I think it would be good if the test started clean, if the package has any activate.d that sets environment variables, Ok, but I wouldn't inherit the environment from the build.sh.

Anyways, I'm going to skip the test section for now, conda-build 2.0 sounds more important.
